docs/sql-distributed-sql-engine-spark-sql-cli.md

```diff
@@ -0,0 +1,198 @@
+        SELECT 1;
+    </code>
+  </td>
+</tr>
+</table>
+
+## Spark SQL CLI Interactive Shell Commands
+
+When `./bin/spark-sql` is run without either the `-e` or `-f` option, it enters interactive shell mode.
+Use `;` (semicolon) to terminate commands. Notice:
+1. The CLI use `;` to terminate commands only when it's at the end of line, and it's not escaped by `\\;`.
+2. `;` is the only way to terminate commands. If the user types `SELECT 1` and presses enter, the console will just wait for input.
+3. If the user types multiple commands in one line like `SELECT 1; SELECT 2;`, the commands `SELECT 1` and `SELECT 2` will be executed separatly.
+4. If `;` appears within a SQL statement (not the end of the line), then it has no special meanings:
+   ```sql
+   -- This is a ; comment
+   SELECT ';' as a;
+   ```
+   This is just a comment line followed by a SQL query which returns a string literal.
+   ```sql
+   /* This is a comment contains ;
+   */ SELECT 1;
+   ```
+   However, if ';' is the end of the line, it terminates the SQL statement. The example above will be terminated into  `/* This is a comment contains ` and `*/ SELECT 1`, Spark will submit these two command and throw parser error (unclosed bracketed comment).
+
+
+<table class="table">
+<tr><th>Command</th><th>Description</th></tr>
+<tr>
+  <td><code>quit</code> or <code>exit</code></td>
+  <td>Exits the interactive shell.</td>
+</tr>
+<tr>
+  <td><code>!&lt;command&gt;</code></td>
+  <td>Executes a shell command from the Spark SQL CLI shell.</td>
+</tr>
+<tr>
+  <td><code>dfs &lt;HDFS dfs command&gt;</code></td>
+  <td>Executes a HDFS <a href="https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HDFSCommands.html#dfs">dfs command</a> from the Spark SQL CLI shell.</td>
+</tr>
+<tr>
+  <td><code>&lt;query string&gt;</code></td>
+  <td>Executes a Spark SQL query and prints results to standard output.</td>
+</tr>
+<tr>
+  <td><code>source &lt;filepath&gt;</code></td>
+  <td>Executes a script file inside the CLI.</td>
+</tr>
+</table>
+
+## Examples
+
+Example of running a query from the command line:
+
+    ./bin/spark-sql -e 'SELECT COL FROM TBL'
+
+Example of setting Hive configuration variables:
+
+    ./bin/spark-sql -e 'SELECT COL FROM TBL' --hiveconf hive.exec.scratchdir=/home/my/hive_scratch
+    
+Example of setting Hive configuration variables and using it in the SQL query:
+
+    ./bin/spark-sql -e 'SELECT ${hiveconf:aaa}' --hiveconf aaa=bbb --hiveconf hive.exec.scratchdir=/home/my/hive_scratch
+    spark-sql> SELECT ${aaa};
+    bbb
+    
+Example of setting Hive variables substitution:
+    
+    ./bin/spark-sql --hivevar aaa=bbb --define ccc=ddd
+    spark-sql> SELECT ${aaa}, ${ccc};
+    bbb ddd
+
+Example of dumping data out from a query into a file using silent mode:
+
+    ./bin/spark-sql -S -e 'SELECT COL FROM TBL' > result.txt
+
+Example of running a script non-interactively from local disk:
+
+    ./bin/spark-sql -f /path/to/spark-sql-script.sql
```

Comment:
```suggestion
Example of running a script non-interactively:

    ./bin/spark-sql -f /path/to/spark-sql-script.sql
```
This should be good enough, as we have already explained how path is interpreted.